File: CleverFerretV2/feature/webfiction/src/main/java/com/cleverferret/v2/feature/webfiction/api/WebfictionFeatureApi.java
package com.cleverferret.v2.feature.webfiction.api;

import com.cleverferret.v2.core.common.result.IntegrationResult;
import com.cleverferret.v2.feature.webfiction.services.webfiction.WebfictionStoryResultV1;

public interface WebfictionFeatureApi {
    String featureKey();

    IntegrationResult<WebfictionStoryResultV1> loadStory(String storyUrl);
}

File: CleverFerretV2/feature/webfiction/src/main/java/com/cleverferret/v2/feature/webfiction/api/WebfictionFeatureApi.kt
package com.cleverferret.v2.feature.webfiction.api

import com.cleverferret.v2.core.common.result.IntegrationResult
import com.cleverferret.v2.feature.webfiction.services.webfiction.WebfictionStoryResultV1

interface WebfictionFeatureApi {
    fun featureKey(): String
    fun loadStory(storyUrl: String): IntegrationResult<WebfictionStoryResultV1>
}
